### ISSUE TITLE
Column header text cut off on scroll

### ISSUE THREAD

**Issue by dkawashima** [pavankataria/SwiftDataTables#54]
iOS 11, most recent SwiftDataTables update.

When I create a table and scroll outside of the initial view, a lot of the header titles get shortened to the point where you cannot tell what the title is.

If all of the headers are visible without having to scroll left, then it works perfectly. But if the datatable is too big to fit without scrolling, I see something like this.

I couldn't find a parameter to change this other than fixed column widths, which shouldn't be necessary here. 
![IMG_1849](https://user-images.githubusercontent.com/10537071/81893875-6e1e4580-9563-11ea-85b7-f59dbed3560f.jpg)

**Comment by pavankataria** [comment]
Okay, I can't reproduce this my end. Could you create a vanilla xcode project so I can check it out please? Aka the bare minimum required to reproduce this please.

I can then take a look at this and see what's going on.

**Comment by dkawashima** [comment]
Cool, thanks! working on getting a vanilla Xcode project. For more info, I've noticed this is only an issue if the value in the column itself is an integer. For strings, it works fine.

**Comment by pavankataria** [comment]
Hi @dkawashima , cheers for the debugging. I look forward to receiving your vanilla project. 

In the mean time I've just upgraded to Catalina and I'm experiencing a wealth of development problems. Xcode keeps crashing and taking me the login screen where I'm then locked out until I restart. Lol, rebooting in safe mode things work well so all hope is not lost. 
It would have been better had I downloaded the latest xcode 11 version that works on Mojave. 

Just informing you all that things might be slow for me in regards to SwiftDataTables work so I appreciate the effort some of you guys have been putting in to make this package more awesome.

**Comment by PerplexedFox** [comment]
> iOS 11, most recent SwiftDataTables update.
> 
> When I create a table and scroll outside of the initial view, a lot of the header titles get shortened to the point where you cannot tell what the title is.
> 
> If all of the headers are visible without having to scroll left, then it works perfectly. But if the datatable is too big to fit without scrolling, I see something like this.
> 
> I couldn't find a parameter to change this other than fixed column widths, which shouldn't be necessary here.
> ![IMG_1849](https://user-images.githubusercontent.com/10537071/81893875-6e1e4580-9563-11ea-85b7-f59dbed3560f.jpg)

@dkawashima, had this problem. You need to change the column width calculation, which is currently quite rough. It works with strings because they are usually wider, which means the "average" width is greater.

**Comment by dkawashima** [comment]
@PerplexedFox do you mean manually changing the column width calculations, for each header?

**Comment by pavankataria** [comment]
@dkawashima @PerplexedFox how would you guys prefer this to be calculated? 
I'm thinking of making it so that the user specifies a hardcoded width for each of the columns which will by default propagate to the header and items width. 

What do you think? I'm considering removing the auto calculation especially since it relies on the data being textual. And iterating through every item in the datasource just to compute the average width for the column seems wasteful.

**Comment by dkawashima** [comment]
yeah i think that makes sense (hardcoded width). otherwise could make sure text doesn't get cut off or resizes to fit current width as Apple typically does with a UILabel. like hardcoded width though

**Comment by pavankataria** [comment]
Sorry for the delay! 
The problem was the internal calculations didn't account for the button indicators. I've updated the width calculations offering different column width strategies for the content while ensuring the headers width will always prevail so the minimum column width despite column strategy will always encompass the header width. In the future perhaps this may not be preferred, for now it makes sense for header text to be visible at all times regardless how small the content is. 

This will be available in the next release and will come with a wealth of performance improvements and features!

**Comment by pavankataria** [comment]
Finally - so this is fixed in v0.9.0. Header text measurement has been improved to properly account for content width.
